rust/worker/src/blockstore/arrow/provider.rs

```diff
@@ -165,19 +196,19 @@ impl BlockManager {
         block.to_block_delta::<KeyWrite::ReadableKey<'new>, ValueWrite::ReadableValue<'new>>(delta)
     }
 
-    pub(super) fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(&self, delta: &BlockDelta) {
+    pub(super) fn commit<K: ArrowWriteableKey, V: ArrowWriteableValue>(
+        &self,
+        delta: &BlockDelta,
+    ) -> Block {
         let record_batch = delta.finish::<K, V>();
         let block = Block::from_record_batch(delta.id, record_batch);
-        self.read_cache.write().insert(block.id, block);
+        block
```

Comment:
do we not want to cache a block if its commited?

Reply:
For the current design, we don't.